#!/usr/bin/env python
# FTC Planner configuration

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t

gen = ParameterGenerator()

gen.add("goal_tolerance", double_t, 0, "A.", 0.2, 0.0, 2.0)

gen.add("angle_tolerance", double_t, 0, "A.", 0.1, 0.0, 6.28)

gen.add("look_ahead_dist", double_t, 0, ".", 0.8, 0, 5.0)

gen.add("max_th", double_t, 0, "Th.", 0.3, 0, 2.0)

gen.add("max_vel", double_t, 0, "Ts.", 0.5, 0, 2.0)

gen.add("max_acc", double_t, 0, "Ts.", 1.0, 0, 10.0)

gen.add("sim_time", double_t, 0, "h.", 0.1, 0, 10.0)

gen.add("frequency", int_t, 0, "freq.", 10, 0, 100)

gen.add("oscllision_angle", double_t, 0, "n.", 0.1, 0, 1.0)

gen.add("safty_padding", double_t, 0, "n.", 1.5, 0, 10.0)

gen.add("over_time", double_t, 0, "n.", 10.0, 1, 600.0)

gen.add("slow_down_factor", double_t, 0, "Factor ", 1, 0, 10.0)

exit(gen.generate("ftc_local_planner", "ftc_local_planner", "FTCPlanner"))
